src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -0,0 +1,1351 @@
+    return array;
+  }
+
+  /**
+   * Generate a bitmap with the specified values set to true. The provided longs values don't have
+   * to be in sorted order, but it may be preferable to sort them from a performance point of view.
+   *
+   * @param dat set values
+   * @return a new bitmap
+   */
+  public static Roaring64NavigableMap bitmapOf(final long... dat) {
+    final Roaring64NavigableMap ans = new Roaring64NavigableMap();
+    ans.add(dat);
+    return ans;
+  }
+
+  /**
+   * Set all the specified values to true. This can be expected to be slightly faster than calling
+   * "add" repeatedly. The provided integers values don't have to be in sorted order, but it may be
+   * preferable to sort them from a performance point of view.
+   *
+   * @param dat set values
+   */
+  public void add(long... dat) {
+    for (long oneLong : dat) {
+      addLong(oneLong);
+    }
+  }
+
+  /**
+   * Add to the current bitmap all longs in [rangeStart,rangeEnd).
+   *
+   * @param rangeStart inclusive beginning of range
+   * @param rangeEnd exclusive ending of range
+   */
+  public void add(final long rangeStart, final long rangeEnd) {
+    int startHigh = high(rangeStart);
+    int startLow = low(rangeStart);
+
+    int endHigh = high(rangeEnd);
+    int endLow = low(rangeEnd);
+
+    for (int high = startHigh; high <= endHigh; high++) {
+      final int currentStartLow;
+      if (startHigh == high) {
+        // The whole range starts in this bucket
+        currentStartLow = startLow;
+      } else {
+        // Add the bucket from the beginning
+        currentStartLow = 0;
+      }
+
+      long startLowAsLong = Util.toUnsignedLong(currentStartLow);
+
+      final long endLowAsLong;
+      if (endHigh == high) {
+        // The whole range ends in this bucket
+        endLowAsLong = Util.toUnsignedLong(endLow);
+      } else {
+        // Add the bucket until the end: we have a +1 as, in RoaringBitmap.add(long,long), the end
+        // is excluded
+        endLowAsLong = Util.toUnsignedLong(-1) + 1;
+      }
+
+      if (endLowAsLong > startLowAsLong) {
+        // Initialize the bitmap only if there is access data to write
+        BitmapDataProvider bitmap = highToBitmap.get(high);
+        if (bitmap == null) {
+          bitmap = new MutableRoaringBitmap();
+          pushBitmapForHigh(high, bitmap);
+        }
+
+        if (bitmap instanceof RoaringBitmap) {
+          ((RoaringBitmap) bitmap).add(startLowAsLong, endLowAsLong);
+        } else if (bitmap instanceof MutableRoaringBitmap) {
+          ((MutableRoaringBitmap) bitmap).add(startLowAsLong, endLowAsLong);
+        } else {
+          throw new UnsupportedOperationException("TODO. Not for " + bitmap.getClass());
+        }
+      }
+    }
+
+    invalidateAboveHigh(startHigh);
+  }
+
+  @Override
+  public LongIterator getReverseLongIterator() {
+    return toIterator(highToBitmap.descendingMap().entrySet().iterator(), true);
+  }
+
+  @Override
+  public void add(long x) {
```

Comment:
Can this replace the method `#addLong(...)`, as this one comes from the provider interface ?

Reply:
I'm uncomfortable with LongBitmapDataProvider.add. One may be confused with the actual meaning of .add(-1) which has a different meaning for BitmapDataProvider and for LongBitmapDataProvider.

What do you thing about rather naming LongBitmapDataProvider.addLong and LongBitmapDataProvider.removeLong ? Then, I would keep Roaring64NavigableMap.addInt (but out of the interface), and remove Roaring64NavigableMap.add

The followingtest fails on the last row, and demonstrate the confusive behavior:

    @Test
	public void testDefaultBehaviorLikeRoaring_MinusOneAsInt() {
		Roaring64NavigableMap longBitmap = newDefaultCtor();
		RoaringBitmap bitmap = new RoaringBitmap();

		longBitmap.add(-1);
		bitmap.add(-1);

		// Ok as -1 === -1
		Assert.assertEquals(bitmap.select(0), longBitmap.select(0));

		// But RoaringBitmap.select has to be interpreted as an unsigned integer: not as an unsigned
		// long
		Assert.assertEquals(Util.toUnsignedLong(bitmap.select(0)), longBitmap.select(0));
	  }